python-manual/modules/ROOT/pages/transactions.adoc

```diff
@@ -251,6 +251,63 @@ def request_inspection(customer_id, other_bank_id, amount, e):
 ----
 
 
+[[process-result]]
+== Process query results
+
+The driver's output of a query is a link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#neo4j.Result[`Result`] object, which does not directly contain the result records.
+Rather, it encapsulates the Cypher result in a rich data structure that requires some parsing on the client side.
+
+When working with a `Result` object, there are two things to keep in mind:
+
+- *The result records are not immediately and entirely fetched and returned by the server*.
+Instead, results come as a _lazy stream_.
+In particular, when the driver receives some records from the server, they are initially _buffered_ in a background queue.
+Records stay in the buffer until they are _consumed_ by the application, at which point they are _removed from the buffer_.
+When no more records are available, the result is _exhausted_.
+- *The result acts as a _cursor_*.
+This means that there is no way to retrieve a previous record from the stream, unless you saved it in an auxiliary data structure.
+
+**The easiest way of processing a result is by casting it to list**, which yields a list of link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#neo4j.Record[`Record`] objects.
+Otherwise, a `Result` object implements a number of methods for processing records.
+The most commonly needed ones are listed below.
+
+// .An essential list of `Result` methods
+[cols="2,3"]
+|===
+|Name |Description
+
+|`value(key=0, default=None)`
+|Return the remainder of the result as a list.
+If `key` is specified, only the given property is included, while `default` allows to specify a value for nodes lacking that property.
+
+|`fetch(n)`
+|Return up to `n` records from the result.
+
+|`single(strict=False)`
+a|Return the next and only remaining record, or `None`.
+Calling this method always exhausts the result.
+
+If more than one record is available,
+
+- `strict==False` -- a warning is generated and the first of these is returned;
+- `strict==True` -- a `ResultNotSingleError` is raised.
```

Comment:
I think this description is misleading. Calling `Result.single(strict=True)` on an empty (or exhausted) stream seems to return `None` according to this. However, it will actually cause a `ResultNotSingleError`.

Reply:
Good point. Addressed trying to keep it as short as possible.